tests/integration_test/http1_deadline_test.go

```diff
@@ -180,3 +186,96 @@ func TestDeadlinesForLocalBackend(t *testing.T) {
 		})
 	}
 }
+
+// Tests the stream idle timeouts configured via deadline in backend rules for a HTTP/1.x backend.
+func TestIdleTimeoutsForUnaryRPCs(t *testing.T) {
+	t.Parallel()
+
+	testData := []struct {
+		desc           string
+		confArgs       []string
+		reqDuration    time.Duration
+		deadlineToTest ConfiguredDeadline
+		wantErr        string
+	}{
+		// Please be cautious about adding too many time-based tests here.
+		// This can slow down our CI system if we sleep for too long.
+		{
+			desc: "When deadline is NOT specified, default deadline (15s) kicks in and the request fails.",
+			confArgs: append([]string{
+				"--stream_idle_timeout=25s",
+			}, utils.CommonArgs()...),
+			reqDuration:    time.Second * 20,
+			deadlineToTest: Default,
+			wantErr:        `504 Gateway Timeout, {"code":504,"message":"upstream request timeout"}`,
+		},
+		{
+			desc: "When deadline is NOT specified, ESPv2 does not honor the global idle timeout flag if the value is lower than the default deadline (15s). Request still succeeds.",
+			confArgs: append([]string{
+				"--stream_idle_timeout=5s",
+			}, utils.CommonArgs()...),
+			reqDuration:    time.Second * 10,
+			deadlineToTest: Default,
+		},
+		{
+			desc: "When a large deadline is specified, it overrides the global stream idle timeout specified by flag. The request succeeds.",
+			confArgs: append([]string{
+				"--stream_idle_timeout=1s",
+			}, utils.CommonArgs()...),
+			reqDuration:    time.Second * 2,
+			deadlineToTest: Short,
+		},
+		{
+			desc: "When a small deadline is specified, it overrides the larger global stream idle timeout specified by flag. The request fails with a 504, not 408.",
+			confArgs: append([]string{
+				"--stream_idle_timeout=15s",
+			}, utils.CommonArgs()...),
+			reqDuration:    time.Second * 8,
+			deadlineToTest: Short,
+			wantErr:        `504 Gateway Timeout, {"code":504,"message":"upstream request timeout"}`,
+		},
+		{
+			desc: "When a small deadline is specified, it overrides the smaller global stream idle timeout specified by flag. The request fails with a 504, not 408.",
```

Comment:
Can we have a test getting 408?

Reply:
It's not possible for that to occur, the behavior clients see is:
- For unary methods: Always return 504 when timeout occurs. We always set stream idle timeout to be more than the response deadline, so 408 will never occur.
- For streaming methods: Always return 408 when timeout occurs. Response deadlines do not apply to streaming methods, so 504 can never occur.

We have tests for these cases split across the 2 integration test files.